internal/k8s/controllers/http_route_controller.go

```diff
@@ -53,5 +61,73 @@ func (r *HTTPRouteReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 func (r *HTTPRouteReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&gateway.HTTPRoute{}).
+		Watches(
+			&source.Kind{Type: &gateway.ReferencePolicy{}},
+			handler.EnqueueRequestsFromMapFunc(r.referencePolicyToRouteRequests),
+		).
 		Complete(gatewayclient.NewRequeueingMiddleware(r.Log, r))
 }
+
+// For UpdateEvents which contain both a new and old object, this transformation
+// function is run on both objects and both sets of Requests are enqueued.
+//
+// This is needed to reconcile any objects matched by both current and prior
+// state in case a ReferencePolicy has been modified to revoke permission from a
+// namespace or to a service
+//
+// It may be possible to improve performance here by filtering Routes by
+// BackendRefs selectable by the To fields, but currently we just revalidate
+// all Routes allowed in the From Namespaces
+func (r *HTTPRouteReconciler) referencePolicyToRouteRequests(object client.Object) []reconcile.Request {
```

Comment:
So this will invalidate a route that was previously validate if a ReferencePolicy is deleted and the cross-namespace route is no longer permitted, right? We had talked about this in the last kube-sig meeting but I don't remember where exactly we landed.

Reply:
the flow for this should be that the re-triggered route will wind up going through the whole resolution/binding shebang again, this time hitting the `RefNotPermitted` case:

https://github.com/hashicorp/consul-api-gateway/blob/df21465e8cb1932786a116052aff6deba233dae2/internal/k8s/reconciler/route.go#L209

Reply:
will wanna make sure this is documented somewhere

Reply:
Yea, this is the big caveat I called out as needing review - with our current logic any BackendRef becoming unpermitted will cause the entire Route to be removed from the listener, likely disrupting traffic. The upstream conformance tests discussion seems to be leaning heavily towards accepting partially-invalid routes and just serving 503 responses for the invalid backends to handle this scenario more safely.

Reply:
🤦‍♂️ I see that called out in the PR body now. Thanks!

Seems like we're getting a little more pressure on handling our own xDS every day lately